core/domain/skill_services.py

```diff
@@ -514,3 +515,107 @@ def save_skill_summary(skill_summary):
     )
 
     skill_summary_model.put()
+
+
+def create_user_skill_mastery(user_id, skill_id, degree_of_mastery):
+    """Creates skill mastery of a user.
+
+    Args:
+        user_id: str. The user id of the logged in user.
+        skill_id: str. The unique id of the skill.
+        degree_of_mastery: float. The degree of mastery of user in the skill.
+    """
+    model_id = user_models.UserSkillMasteryModel.get_model_id(
+        user_id, skill_id)
+
+    user_skill_mastery = skill_domain.UserSkillMastery(
+        model_id, user_id, skill_id, degree_of_mastery)
+    save_user_skill_mastery(user_skill_mastery)
+
+
+def save_user_skill_mastery(user_skill_mastery):
+    """Stores skill mastery of a user.
+
+    Args:
+        user_skill_mastery: dict. The user skill mastery model of a user.
+    """
+    user_skill_mastery_model = user_models.UserSkillMasteryModel(
```

Comment:
Create the model_id on the fly here. Don't store it in the domain object.

Reply:
ok